kong/plugins/basic-auth/api.lua

```diff
@@ -1,9 +1,41 @@
 local crud = require "kong.api.crud_helpers"
+local basic_auth_utils = require "kong.plugins.basic-auth.utils"
+
+local function get_config(dao_factory)
+  -- in case of no conf.
+  local default_config = {
+    hide_credentials = false,
+    encryption_method = "plain",
+  }
+
+  local data, err = dao_factory.plugins:find_by_keys({ name = "basic-auth" })
```

Comment:
exactly, i was hitting that wall, (_this code just retrive the first plugin conf_) wanted someone that knows the internals to advice.
currently there is only one encryption method (plain). 
i thought it will be more flexible to have encryption method along side others, so that, we can be smoothly backward compatible, and a consumer/api (not sure what entity) could have different encryption methods.

do you see how this link could be achieved ?